configuration_go/abstr/kubernetes/cmdoption/cmdopt.go

```diff
@@ -0,0 +1,228 @@
+package cmdopt
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"strings"
+)
+
+const (
+	tagName              = "opt"
+	noValModifier        = "noval"
+	singleHyphenModifier = "single-hyphen"
+)
+
+type CmdOptions struct{}
+
+// GetOpts is used to generate command line options from a struct
+// by mapping the fields values to the option name defined in the opt tag (first position).
+// GetOpts returns a slice of strings, each string representing an option.
+// Following types are supported: string, int, bool, float64, time.Duration, slice of supported types, sub struct implementing the Stringer interface.
+// Pointer types are used if not nil. Private fields are ignored.
+// Additional tags can be added to the opt tag, separated by a comma, to modify its default behavior:
+// - noval: the option is added without a value if the field is true.
+// - single-hyphen: the option is prefixed with a single hyphen instead of a double hyphen.
+func GetOpts(obj interface{}) []string {
```

Comment:
Ok, first off I think this is a really cool way of getting the flags from a struct, and I see it being super clean approach. I'd actually love to see some form of CLI package with this at some point. 🙂 

But, this is still a maintenance burden in the context of Observatorium. I'm keen to merge this, but with a TODO, to upstream this in the future, so that this config is derived from the same place as where the original struct is defined (i.e in Thanos).

Reply:
Cool! But I don't know if it will ever make it to the other side (parsing args and map them to the struct). There are probably better libs for that.

For now, I don't see how we can escape the maintenance burden of these flags. Because either we provide a tailored API for these components like what is done in `query` component. But we still have to add new methods when new flags appear. Either we import directly from Thanos, but this may take a while before being possible.

Also, if a flag is not yet listed here but present on Thanos, you can still add it as an ExtraOpt or manually in the container.Args.